portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/clay/ClayAlertToast.testcase

```diff
@@ -97,6 +97,31 @@ definition {
 		}
 	}
 
+	@description = "LPS-170577 - Verify toast alert with action button disappears after 10 seconds"
+	@priority = "5"
+	test DisappearsAfter10SecsWithActions {
+		task ("When a toast alert appears") {
+			Click(
+				key_text = "Disappears After 10 Seconds",
+				locator1 = "Button#ANY");
+		}
+
+		task ("And there is an action button on it") {
+			VerifyElementPresent(locator1 = "//*[@id='ToastAlertContainer']//*[contains(@class, 'alert-info')]//a");
+		}
+
+		task ("Then the alert disappears after 10 secs") {
+
+			// For LPS-170577, need this pause to see if after 10 seconds, the alert is not showing anymore
+
+			Pause(locator1 = "10000");
+
+			AssertElementNotPresent.assertElementNotPresent(
+				key_type = "info",
+				locator1 = "ClayAlertToast#ALERT_TOAST");
+		}
```

Comment:
AssertElementPresent function has a wait for. We don't want the wait for, we want to immediately check if the toast alert has disappeared. I'm pretty sure if you put a 15 second pause the test will pass (false positive).

I think something like the following would immediately check if the element is not present:

```
if (!(IsElementPresent.NoSPARefresh(locator1 = "ClayAlertToast#ALERT_TOAST"))) {
			TestUtils.pass(message = "Test passed: Alert toast with action disappeared after 10 seconds");
		}
		else {
			fail("Test failed: Clay Toast alert with action is still present after 10 seconds.");
		}
```

Could you try this out?

Reply:
cc/ @mateusralv fyi

Reply:
done ✅